### PR TITLE
Prisma cloud vm prioritization

### PR DESCRIPTION
**Pull request by tomer-pan**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Adding content for "VM Alert Prioritization" use case.

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [x] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

### CODE REVIEW

File: Packs/PrismaCloud/Playbooks/playbook-Prisma_Cloud_-_VM_Alert_Prioritization.yml [hunk 1]
```diff
@@ -0,0 +1,1492 @@
...
+
+  The flow of this playbook is as following:
+  1) Check the CSP type.
+  2) Check whether there's a public IP associated to the instance. In case there is, it will continue to other steps, if not, it will set the severity of the incident to "Low" and will close the incident.
+  3) Check if there are any vulnerabilities or findings related to the instance.
+  4) Check if there are any IAM permissions associated to the instance.
+  5) Set the incident severity based on the results:
+    - LOW - No public IP was found.
+    - Medium - Public IP was found, other checks didn't return results.
+    - High - Public IP was found and also on of the other checks returned results.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Playbooks/playbook-Prisma_Cloud_-_VM_Alert_Prioritization.yml:15]:
```suggestion
    - High - Public IP was found and also one of the other checks returned results.
```

File: Packs/PrismaCloud/Playbooks/playbook-Prisma_Cloud_-_VM_Alert_Prioritization.yml [hunk 2]
```diff
@@ -0,0 +1,1492 @@
...
+    - Critical - Public IP was found and bot of the other checks returned results.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Playbooks/playbook-Prisma_Cloud_-_VM_Alert_Prioritization.yml:16]:
```suggestion
    - Critical - Public IP was found and both of the other checks returned results.
```

File: Packs/PrismaCloud/Playbooks/playbook-Prisma_Cloud_-_VM_Alert_Prioritization.yml [hunk 3]
```diff
@@ -0,0 +1,1492 @@
...
+    isautoswitchedtoquietmode: false
+  "26":
+    id: "26"
+    taskid: 8d1853cb-4627-4bc8-87e3-2cf9e94389e0
+    type: condition
+    task:
+      id: 8d1853cb-4627-4bc8-87e3-2cf9e94389e0
+      version: -1
+      name: Was a public IP found?
+      description: ' '
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Playbooks/playbook-Prisma_Cloud_-_VM_Alert_Prioritization.yml:896]: Was this supposed to be " ?

File: Packs/PrismaCloud/Playbooks/playbook-Prisma_Cloud_-_VM_Alert_Prioritization.yml [hunk 4]
```diff
@@ -0,0 +1,1492 @@
...
+    task:
+      id: ba8ba196-85ed-489e-838c-c25865beeaf7
+      version: -1
+      name: Format IAM Permissions as markdown table
+      description: |-
+        Set a value in context under the key you entered. If no value is entered, the script doesn't do anything.
+
+        This automation runs using the default Limited User role, unless you explicitly change the permissions.
+        For more information, see the section about permissions here:
+        https://docs.paloaltonetworks.com/cortex/cortex-xsoar/6-2/cortex-xsoar-admin/playbooks/automations.html
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Playbooks/playbook-Prisma_Cloud_-_VM_Alert_Prioritization.yml:1052]:
```suggestion
        https://docs-cortex.paloaltonetworks.com/r/Cortex-XSOAR/6.10/Cortex-XSOAR-Administrator-Guide/Automations
```

File: Packs/PrismaCloud/Playbooks/playbook-Prisma_Cloud_-_VM_Alert_Prioritization.yml [hunk 5]
```diff
@@ -0,0 +1,1492 @@
...
+    task:
+      id: ed51c23d-1794-4411-8fb2-2fa328742ed7
+      version: -1
+      name: Format findings as markdown table
+      description: |-
+        Set a value in context under the key you entered. If no value is entered, the script doesn't do anything.
+
+        This automation runs using the default Limited User role, unless you explicitly change the permissions.
+        For more information, see the section about permissions here:
+        https://docs.paloaltonetworks.com/cortex/cortex-xsoar/6-2/cortex-xsoar-admin/playbooks/automations.html
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Playbooks/playbook-Prisma_Cloud_-_VM_Alert_Prioritization.yml:1103]:
```suggestion
        https://docs-cortex.paloaltonetworks.com/r/Cortex-XSOAR/6.10/Cortex-XSOAR-Administrator-Guide/Automations
```

File: Packs/PrismaCloud/Playbooks/playbook-Prisma_Cloud_-_VM_Alert_Prioritization.yml [hunk 6]
```diff
@@ -0,0 +1,1492 @@
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "38":
+    id: "38"
+    taskid: 31a1f2ba-107a-4da3-80f9-9f9928e850fc
+    type: condition
+    task:
+      id: 31a1f2ba-107a-4da3-80f9-9f9928e850fc
+      version: -1
+      name: Send a slack message?
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Playbooks/playbook-Prisma_Cloud_-_VM_Alert_Prioritization.yml:1403]:
```suggestion
      name: Send a Slack message?
```

File: Packs/PrismaCloud/Playbooks/playbook-Prisma_Cloud_-_VM_Alert_Prioritization.yml [hunk 7]
```diff
@@ -0,0 +1,1492 @@
...
+- key: JiraIssueTypeName
+  value: {}
+  required: false
+  description: The name of the issue type to be opened.
+  playbookInputQuery:
+- key: CloseIncidentAfterNotification
+  value:
+    simple: "True"
+  required: false
+  description: Whether to close the incident after jira/slack issue was opened. Default value is "True".
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Playbooks/playbook-Prisma_Cloud_-_VM_Alert_Prioritization.yml:1477]:
```suggestion
  description: Whether to close the incident after Jira/Slack issue was opened. Default value is "True".
```

File: Packs/PrismaCloud/Playbooks/playbook-Prisma_Cloud_-_VM_Alert_Prioritization_README.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+The flow of this playbook is as following:
+1) Check the CSP type.
+2) Check whether there's a public IP associated to the instance. In case there is, it will continue to other steps, if not, it will set the severity of the incident to "Low" and will close the incident.
+3) Check if there are any vulnerabilities or findings related to the instance.
+4) Check if there are any IAM permissions associated to the instance.
+5) Set the incident severity based on the results:
+  - LOW - No public IP was found.
+  - Medium - Public IP was found, other checks didn't return results.
+  - High - Public IP was found and also on of the other checks returned results.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Playbooks/playbook-Prisma_Cloud_-_VM_Alert_Prioritization_README.md:11]:
```suggestion
  - High - Public IP was found and also one of the other checks returned results.
```

File: Packs/PrismaCloud/Playbooks/playbook-Prisma_Cloud_-_VM_Alert_Prioritization_README.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+  - Critical - Public IP was found and bot of the other checks returned results.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Playbooks/playbook-Prisma_Cloud_-_VM_Alert_Prioritization_README.md:12]:
```suggestion
  - Critical - Public IP was found and both of the other checks returned results.
```

File: Packs/PrismaCloud/Playbooks/playbook-Prisma_Cloud_-_VM_Alert_Prioritization_README.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+
+## Playbook Inputs
+
+---
+
+| **Name** | **Description** | **Default Value** | **Required** |
+| --- | --- | --- | --- |
+| JiraProjectName | The project name with which to associate the issue. |  | Optional |
+| JiraIssueTypeName | The name of the issue type to be opened. |  | Optional |
+| CloseIncidentAfterNotification | Whether to close the incident after jira/slack issue was opened. | True | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Playbooks/playbook-Prisma_Cloud_-_VM_Alert_Prioritization_README.md:52]:
```suggestion
| CloseIncidentAfterNotification | Whether to close the incident after Jira/Slack issue was opened. | True | Optional |
```

File: Packs/PrismaCloud/ReleaseNotes/2_1_0.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+  * Instance affected by OpenSSL X.509 email address 4-Byte BOF (Spooky SSL) vulnerability is exposed to network traffic from the internet [CVE-2022-3602]
+  * Instance affected by Samba vfs_fruit module remote code execution vulnerability is exposed to network traffic from the internet [CVE-2021-44142]
+  * Instance affected by SMB DCE/RPC remote code execution vulnerability is exposed to network traffic from the internet [CVE-2022-26809]
+  * Instance affected by Spring Cloud Function SpringShell vulnerability is exposed to network traffic from the internet [CVE-2022-22963]
+  * Instance affected by Spring Framework SpringShell vulnerability is exposed to network traffic from the internet [CVE-2022-22965]
+  * Instance affected by Text4shell RCE vulnerability is exposed to network traffic from the internet [CVE-2022-42889]
+  * Instance is communicating with ports known to mine Bitcoin
+  * Instance is communicating with ports known to mine Ethereum
+  * Instances exposed to network traffic from the internet
+  * Instances exposed to network traffic from the internet
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/ReleaseNotes/2_1_0.md:30]: This is the same as the previous line.

File: Packs/PrismaCloud/ReleaseNotes/2_1_0.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+
+##### Prisma Cloud - Incoming Mapper
+
+- Added mapping for the new "VM Alert Prioritization" incident type.
+
+#### Playbooks
+
+##### New: Prisma Cloud - VM Alert Prioritization
+
+- This playbook will handle incidents related to Prisma Cloud VM alerts such as "Instances exposed to network traffic from the internet".
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/ReleaseNotes/2_1_0.md:59]:
```suggestion
This playbook will handle incidents related to Prisma Cloud VM alerts such as "Instances exposed to network traffic from the internet".
```

File: Packs/PrismaCloud/ReleaseNotes/2_1_0.md [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+
+#### Incident Types
+
+- New: **Prisma Cloud - VM Alert Prioritization**
+
+#### Layouts
+
+##### New: Prisma Cloud - VM Alert Prioritization
+
+- Added a new layout for the new "VM Alert Prioritization" incident type (Available from Cortex XSOAR 6.8.0).
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/ReleaseNotes/2_1_0.md:47]:
```suggestion
Added a new layout for the new "VM Alert Prioritization" incident type (Available from Cortex XSOAR 6.8.0).
```

File: Packs/PrismaCloud/ReleaseNotes/2_1_0.md [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+
+#### Mappers
+
+##### Prisma Cloud - Incoming Mapper
+
+- Added mapping for the new "VM Alert Prioritization" incident type.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/ReleaseNotes/2_1_0.md:53]:
```suggestion
Added mapping for the new "VM Alert Prioritization" incident type.
```

File: Packs/PrismaCloud/ReleaseNotes/2_1_0.md [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+
+The flow of this playbook is as following:
+1) Check the CSP type.
+2) Check whether there's a public IP associated to the instance. In case there is, it will continue to other steps, if not, it will set the severity of the incident to "Low" and will close the incident.
+3) Check if there are any vulnerabilities or findings related to the instance.
+4) Check if there are any IAM permissions associated to the instance.
+5) Set the incident severity based on the results:
+  - LOW - No public IP was found.
+  - Medium - Public IP was found, other checks didn't return results.
+  - High - Public IP was found and also on of the other checks returned results.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/ReleaseNotes/2_1_0.md:69]:
```suggestion
  - High - Public IP was found and also one of the other checks returned results.
```

File: Packs/PrismaCloud/ReleaseNotes/2_1_0.md [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+  - Critical - Public IP was found and bot of the other checks returned results.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/ReleaseNotes/2_1_0.md:70]:
```suggestion
  - Critical - Public IP was found and both of the other checks returned results.
```

File: Packs/PrismaCloud/ReleaseNotes/2_1_0.md [hunk 17]
```diff
@@ -0,0 +1,72 @@
+
+#### Classifiers
+
+##### Prisma Cloud - Classifier
+
+- Classifying the following alerts to the new "VM Alert Prioritization" incident type:
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/ReleaseNotes/2_1_0.md:6]:
```suggestion
Classifying the following alerts to the new "VM Alert Prioritization" incident type: 
```

**Comment by ShirleyDenkberg** [comment]
@michalgold Doc review completed.

**Review by idovandijk — COMMENTED** [review]
Looks great! A few comments
- Please also add associations for existing fields to the incident type you've added.
Remember that for existing fields you must add associated **both** to `associatedTypes` **and** to `systemAssociatedTypes` in order for it to work.
- Note that the classifiers are from 6.0.0 but the rest of the content is from 6.8.0. I think that if the classifier is using fields that don't exist on the system it will just ignore them, but worth verifying with dev as well.

Let me know once you're done and I'll approve :)

**Review by idovandijk — APPROVED** [review]
Awesome

**Pull request merged**